tests/test_middleware_fastapi.py

```
@@ -0,0 +1,327 @@
+        with _https_client(app) as tc:
+            resp = tc.get("/login?idp_initiated_login=some.jwt.token", follow_redirects=False)
+
+        self.assertEqual(resp.status_code, 302)
+        client.get_idp_initiated_login_claims.assert_called_once_with("some.jwt.token")
+        call_options = client.get_authorization_url.call_args[0][1]
+        self.assertEqual(call_options.connection_id, "conn_123")
+        self.assertEqual(call_options.organization_id, "org_456")
+        self.assertEqual(call_options.login_hint, "user@example.com")
+
+    def test_login_with_invalid_idp_initiated_login_falls_back_to_normal_login(self):
+        app, auth, client = _build_app()
```

Ruff (0.16.1)

[warning] Unpacked variable `auth` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)